packages/near-fast-auth-signer-e2e-tests/tests/devices.spec.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  KeyPair,
+} from '@near-js/crypto';
+import { expect, test } from '@playwright/test';
+
+import PageManager from '../pages/PageManager';
+import {
+  createAccount, deleteAccount, generateKeyPairs, initializeAdmin,
+  isServiceAccountAvailable
+} from '../utils/createAccount';
+import { getRandomEmailAndAccountId } from '../utils/email';
+import { setupPasskeysFunctions } from '../utils/passkeys';
+
+let testUserUid;
+
+test.beforeAll(async () => {
+  if (isServiceAccountAvailable()) {
+    initializeAdmin();
+  }
+});
+
+test('device page delete existing keys and continue sign in', async ({ page, baseURL }) => {
+  test.skip(!isServiceAccountAvailable(), 'Skipping test due to missing service account');
+
+  const pm = new PageManager(page);
+  test.setTimeout(120000);
+  const { email, accountId } = getRandomEmailAndAccountId();
+
+  await page.goto(baseURL);
+  const walletSelector = page.locator('#ws-loaded');
+  await walletSelector.waitFor();
+
+  const oidcKeyPair = KeyPair.fromRandom('ED25519');
+
+  // As of 14 May 2024, creating an account with 5 keypairs will be just enough to redirected to devices page
+  const keypairs = generateKeyPairs(5);
+  const {
+    createAccountResponse,
+    userUid
+  } = await createAccount({
+    email,
+    accountId,
+    oidcKeyPair,
+    keypairs,
+  });
+
+  // will be used to delete account
+  testUserUid = userUid;
+
+  expect(createAccountResponse.ok).toBe(true);
+
+  await setupPasskeysFunctions(page, 'page', {
+    isPassKeyAvailable:  true,
+    keyPairForCreation:  oidcKeyPair,
+    keyPairForRetrieval: oidcKeyPair,
+    shouldCleanStorage:  false
+  });
+
+  await pm.getLoginPage().signInWithEmail(email);
+  await page.waitForLoadState('domcontentloaded');
```

Comment:
Nothing against, asking more to understand. Why you are using this check here? 

On my tests I use this validation `await pm.getEmailPage().hasLoaded();` before call the `pm.getAuthCallBackPage().handleEmail`

Reply:
Oh was just wanted to simply sign in wait until everything show up, your recommended changes also work just fine so happy to replace

Reply:
Cool, thanks Danny, will resolve when the change is applied